tp_framework/core/utils.py

```diff
@@ -40,14 +42,27 @@ def list_pattern_instances_by_pattern_id(language: str, pattern_id: int, tp_lib_
         raise ee
 
 
+def get_tpi_id_from_jsonpath(jp: Path) -> int:
+    return get_id_from_name(jp.parent.name)
+
 def get_pattern_dir_from_id(pattern_id: int, language: str, tp_lib_dir: Path) -> Path:
     tp_lib_dir_lang_dir: Path = tp_lib_dir / language
     if tp_lib_dir_lang_dir.is_dir():
-        return list(filter(lambda p: int(p.name.split("_")[0]) == pattern_id, tp_lib_dir_lang_dir.iterdir()))[0]
+        return list(filter(lambda p: get_id_from_name(p.name) == pattern_id, list_dirs_only(tp_lib_dir_lang_dir)))[0]

```

Comment:
```suggestion
       pattern_with_id = list(filter(lambda p: get_id_from_name(p.name) == pattern_id, list_dirs_only(tp_lib_dir_lang_dir)))
        if pattern_with_id:
            return pattern_with_id[0]
    raise PatternDoesNotExists(pattern_id)
```
The if statement only checks if the `tp_lib_dir_lang_dir` exists. But what if the wanted pattern ID in the directory does not exist? The current version will throw a `KeyError` there. I think a `PatternDoesNotExist` Exception is a little bit more descriptive and additionally, this matches the result expected in `qualitytests/core/test_utils.py`